classes/form/CustomerAddressForm.php

```diff
@@ -68,8 +68,23 @@ public function fillWith(array $params = [])
 
     public function validate()
     {
+        $validated = true;
+
         if (!parent::validate()) {
-            return false;
+            $validated =  false;
+        }
+
+        if (($city = $this->getField('city'))) {
+            if (isset(Address::$definition['fields']['city']['size']) &&
+                strlen($city->getValue()) > Address::$definition['fields']['city']['size']) {
+                $city->addError(sprintf(
+                    $this->translator->trans(
+                        'city name is too long - should have a maximum of %d characters', [], 'Address'
```

Comment:
Try without this validation and you'll see :wink: 

It would normally only use `Validate::isCityName()` for the city field, but doesn't take the length into account, resulting in an exception being thrown.